internal/watch/file_watcher_test.go

```diff
@@ -67,3 +69,73 @@ func TestFileWatcher(t *testing.T) {
 		t.Fatalf("expected an event shortly after writing a file")
 	}
 }
+
+func TestFileWatcherWithNestedSymlink(t *testing.T) {
+	target1, err := ioutil.TempFile("", "t1")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	defer target1.Close()
+	defer os.Remove(target1.Name())
+	dir := path.Dir(target1.Name())
+
+	innerLink := path.Join(dir, "innerLink")
+	if err = os.Symlink(target1.Name(), innerLink); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	defer os.Remove(innerLink)
+	mainLink := path.Join(dir, "mainLink")
+	if err = os.Symlink(innerLink, mainLink); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	defer os.Remove(mainLink)
+
+	targetName, err := filepath.EvalSymlinks(mainLink)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	count := 0
+	events := make(chan bool, 10)
+	fw, err := NewFileWatcher(mainLink, func() {
+		count++
+		if count != 1 {
+			t.Fatalf("expected 1 but returned %v", count)
+		}
+		if targetName, err = filepath.EvalSymlinks(mainLink); err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+		events <- true
+	})
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	defer fw.Close()
+
+	if targetName != target1.Name() {
+		t.Fatalf("expected symlink to point to %v, not %v", target1.Name(), targetName)
+	}
```

Comment:
I'd move this assertion right below line 96. By reading the code I thought this assertion somehow depends on what is happening between line 97 and 113, but seems like they are unrelated.

Reply:
I put it here because I wanted to wait until everything was set up (including the fileWatcher itself) to assert this, but I'm fine to move it to line 97. I should hope the creation of the watcher doesn't do anything weird :)